packages/cloud-config/oem/00_rootfs.yaml

```diff
@@ -8,19 +8,26 @@
 name: "Rootfs Layout Settings"
 stages:
   rootfs.before:
-    - name: "Layout configuration"
+    - name: "Pull data from provider"
+      datasource:
+        providers: ["aws", "gcp", "openstack", "cdrom"]
```

Comment:
could this datasource somehow affect the partition or expect the partition layout? I.E in the case of cloud-config you can execute stuff

Reply:
I am not sure I understand the question. The datasource might introduce new cloud config that might call the `layout` plugin and create or expand partitions. The default in-image config does not include anything like changing disk layout and it assumes a COS_OEM for the recovery system and COS_OEM and COS_PERSISTENT for the active and passive systems. The idea is that datasource is pulled at first stage of yip so **any** stage is valid in datasource, even `rootfs.before` can be called from data sources since `/system/oem` has precedence comapred with `/oem`.